ocaml/xapi/quicktest_import_raw_vdi.ml

```diff
@@ -0,0 +1,55 @@
+(*
+ * Copyright (C) 2017 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+
+open Client
+open Quicktest_common
+open Http
+
+let http request f =
+  let open Xmlrpc_client in
+  let transport =
+    if !using_unix_domain_socket
+    then Unix Xapi_globs.unix_domain_socket
+    else SSL(SSL.make ~use_fork_exec_helper:false (), !host, 443) in
+  with_transport transport (with_http request f)
+
+let import_raw_vdi ~session_id ~task_id f =
+  let req = Http.Request.make ~version:"1.0"
+      ~user_agent:"quicktest"
+      ~query:[
+        "session_id", Ref.string_of session_id;
+        "sr_id","fake_sr";
+        "task_id", Ref.string_of task_id
+      ]
+      Http.Put "/import_raw_vdi" in
+  http req (fun (_, fd) -> f fd)
```

Comment:
I realised that we could also check that we return a correct HTTP response code here, I think that's the first parameter to this function: `fun (response, _fd) -> f response`, and then below in `let start` we could write:
```ocaml
let response = ref None in
...
import_raw_vdi ~session_id ~task_id (fun r -> response := Some r);
...
with e ->
...
  if a.API.task_status = `failure && (response.Http.Response.code = "400")
...
```
(I didn't check that this code compiles.)